docs/topics/media-pipeline.rst

```diff
@@ -127,9 +128,59 @@ Whose ``SHA1 hash`` is::
 
     3afec3b4765f8f0a07b78f98c07b83f013567a0a
 
-Will be downloaded and stored in the following file::
+Will be downloaded and stored in your chosen :ref:`storage method <topics-supported-storage>` using the following file name::
+
+   3afec3b4765f8f0a07b78f98c07b83f013567a0a.jpg
+
+Custom File Naming
+-------------------
+You may wish to use a different calculated file name for saved files.
+For example, classifying an image by including meta in the file name.
+
+Customize file names by overriding the ``file_path`` method of your
+media pipeline.
+
+For example, an image pipeline with image url::
+
+   http://www.example.com/product/images/large/front/0000000004166
+
+Can be processed into a file name with a condensed hash and the perspective
+``front``::
```

Comment:
```suggestion
Can be stored as::
```

Reply:
This change removes signaling of the specific intent of the code. I chose this example because retail products so often provide multiple perspectives, i.e. front, back, left, right. And Scrapy seems to often be used for processing these images.

If this sentence is unclear or too wordy, can you suggest an alternative way to keep this "result" information?

Reply:
OK. I did let myself go with the feedback :sweat_smile:

Reply:
:) The PR is very much improved from the feedback. Thank you.